### DISCUSSION TITLE
Curso Fullstack .Net - Aula Endpoints

### DISCUSSION THREAD

**Discussion by danilobicas-unoeste** [balta-io/forum#610] [Tira Dúvidas] [answered]
Ali nos exemplos foram mostrados endpoints GET onde eu retorno todas as categorias e passo pagesize e pagenumber [FROMQuery]..... E no caso se eu tiver que passar mais opções, exemplo, suponha que eu queira buscar todas as transações mas tenha opções de busca como: período de datas e somente as transações de débito ou outras opções: por período de data e por banco de origem.... Eu penso que poderia passar via post um objeto com todas as opções inclusive o pagesize e o pagenumber..... Poderia ser assim? Como ficaria isso???

Obrigado

Atte.

**Comment by henriqueweege** [comment, marked as answer]
Oi, @danilobicas-unoeste ! Tudo certo?
É isso mesmo! Você pode ter uma classe com a responsabilidade de conter esses parâmetros. Inclusive, você pode usar DataAnnotations para validar campos que estão nessa classe. Vou deixar aqui embaixo um exemplo ;)

Assim ficou a calsse de parâmetros que eu criei (repara o DataAnnotation para validar e-mail ;) ):
![image](https://github.com/balta-io/forum/assets/96916801/1da6db4c-da75-4ecd-bbae-447afbd1ab16)

Assim ficou o end-point onde eu uso a classe:
![image](https://github.com/balta-io/forum/assets/96916801/651a8e17-4883-49c0-8d7c-ac2c3faee19c)

Isso é o que acontece quando passo uma string que não é está no formato de um e-mail (também da para ver nesse print como fica a URL):
![image](https://github.com/balta-io/forum/assets/96916801/cb5dc0d7-0b99-41af-9257-90691256513f)

Caso contrário, bate certinho no método:
![image](https://github.com/balta-io/forum/assets/96916801/c106861e-cc0f-4e4c-b08d-15ac7960b55a)


Valeeeu!
